modules/fedimint-ln-client/src/receive.rs

```diff
@@ -234,21 +234,34 @@ impl LightningReceiveConfirmedInvoice {
 
     async fn transition_funded(
         old_state: LightningReceiveStateMachine,
-        keypair: KeyPair,
+        receiving_key: ReceivingKey,
         result: Result<IncomingContractAccount, LightningReceiveError>,
         dbtx: &mut ClientSMDatabaseTransaction<'_, '_>,
         global_context: DynGlobalClientContext,
     ) -> LightningReceiveStateMachine {
         match result {
             Ok(contract) => {
-                let (txid, out_points) =
-                    Self::claim_incoming_contract(dbtx, contract, keypair, global_context).await;
-                LightningReceiveStateMachine {
-                    operation_id: old_state.operation_id,
-                    state: LightningReceiveStates::Funded(LightningReceiveFunded {
-                        txid,
-                        out_points,
-                    }),
+                match receiving_key {
+                    ReceivingKey::Personal(keypair) => {
+                        let (txid, out_points) =
+                            Self::claim_incoming_contract(dbtx, contract, keypair, global_context)
+                                .await;
+                        LightningReceiveStateMachine {
+                            operation_id: old_state.operation_id,
+                            state: LightningReceiveStates::Funded(LightningReceiveFunded {
+                                txid,
+                                out_points,
+                            }),
+                        }
+                    }
+                    ReceivingKey::External(_) => {
+                        // Claim successful
+                        // fixme is this okay?
+                        LightningReceiveStateMachine {
+                            operation_id: old_state.operation_id,
+                            state: LightningReceiveStates::Success(vec![]),
+                        }
```

Comment:
I looked at `subscribe_ln_receive` and am fairly certain that it would get stuck (but *should* panic imo since it's an internal programming inconsistency) for external receives since it doesn't expect `Success` before `Funded`.

Reply:
think i fixed this by just checking if it is an external receiving and yielding the correct state